app/assets/javascripts/models/ui/reverb_retirement.coffee

```diff
@@ -0,0 +1,49 @@
+#= require util/js.cookies
+#= require util/metrics
+
+ns = @edsc.models.ui
+data = @edsc.models.data
+
+ns.ReverbRetirement = do (ko,
+                          ajax = @edsc.util.xhr.ajax,
+                          config=@edsc.models.data.config) ->
+  class ReverbRetirement
+    constructor: ->
+
+    referrerIsReverb: () =>
+      console.log "Checking referrer: " + document.referrer
+      console.log "Checking referrer against ENV value: " + edsc.config.reverb_url
+      referrer = if document.referrer then document.referrer.match(/:\/\/(.[^/]+)/)[1] else false
+      reverb = if edsc.config.reverb_url then edsc.config.reverb_url.match(/:\/\/(.[^/]+)/)[1] else false
+      return referrer == reverb && referrer
```

Comment:
If ```document.referrer``` is not from reverb but ```edsc.config.reverb_url``` is, will the function return ```true```? In this case, the referrer is apparently not reverb, then why should it return true?

Reply:
>>"If document.referrer is not from reverb but edsc.config.reverb_url is, will the function return true?"

Nope!  ;)

Reply:
what if there is no referrer? so ```referrer``` will be ```false``` and ```reverb``` will be an url. Then ```reverb && referrer``` will be ```false```. So that ```referrer == reverb && referrer``` will be equivalent to ```false == false``` which is ```true```. If my deduction is right, which is probably not the case, the function will think that ```referrerIsReverb``` when there is no referrer at all??

Reply:
If referrer is false then it will return false - same if they are both false - take a closer look at the return conditional (adding parenthesis for emphasis):

return (referrer == reverb) && (referrer)
evaluates to
return (false == false) && (false)
evaluates to
return (true) && (false)
evaluates to - false!  Both sides of the && must be true for it to be true.